src/shims/unix/fd.rs

```diff
@@ -180,14 +197,51 @@ impl FileDescriptor {
         RefMut::map(self.0.borrow_mut(), |fd| fd.as_mut())
     }
 
-    pub fn close<'ctx>(self, communicate_allowed: bool) -> InterpResult<'ctx, io::Result<()>> {
+    pub fn close<'tcx>(
+        self,
+        ecx: &mut MiriInterpCx<'tcx>,
+        communicate_allowed: bool,
+    ) -> InterpResult<'tcx, io::Result<()>> {
         // Destroy this `Rc` using `into_inner` so we can call `close` instead of
         // implicitly running the destructor of the file description.
         match Rc::into_inner(self.0) {
-            Some(fd) => RefCell::into_inner(fd).close(communicate_allowed),
+            Some(fd) => RefCell::into_inner(fd).close(ecx, communicate_allowed),
             None => Ok(Ok(())),
         }
     }
+    pub fn downgrade(&self) -> WeakFileDescriptor {
+        WeakFileDescriptor(Rc::downgrade(&self.0))
+    }
+}
+
+// WeakFileDescriptor is used in epoll ready_list and interest_list to avoid strong references,
+// so the file description can be closed properly.
```

Comment:
This should be a doc comment.

Also, the comment is very confusing. Imagine someone reading this file and never hearing about epoll before. What do they need to know? I don't think it matters where this is used. What matters is what this type does.